src/Symfony/Component/Routing/Annotation/Route.php

```diff
@@ -65,6 +66,8 @@ public function __construct(
             $data = ['path' => $data];
         } elseif (!\is_array($data)) {
             throw new \TypeError(sprintf('"%s": Argument $data is expected to be a string or array, got "%s".', __METHOD__, get_debug_type($data)));
+        } elseif ([] !== $data) {
```

Comment:
0 < \func_num_args()?

Reply:
That would mean that almost any constructor call with arguments would trigger the deprecation:

https://3v4l.org/l542H

Reply:
only if `\is_array($data)`

Reply:
OH, got it! :)